src/cart/crt_internal_types.h

```diff
@@ -32,286 +31,277 @@ struct crt_hg_gdata;
+	char   *noc_auth_key;
 	/* IP addr str for the noc_interface */
-	char		 noc_ip_str[INET_ADDRSTRLEN];
+	char    noc_ip_str[INET_ADDRSTRLEN];
 };
 
 struct crt_prov_gdata {
 	/** NA plugin type */
-	int			cpg_provider;
+	int                  cpg_provider;
 
-	struct crt_na_config	cpg_na_config;
+	struct crt_na_config cpg_na_config;
 	/** Context0 URI */
-	char			cpg_addr[CRT_ADDR_STR_MAX_LEN];
+	char                 cpg_addr[CRT_ADDR_STR_MAX_LEN];
 
 	/** CaRT contexts list */
-	d_list_t		cpg_ctx_list;
+	d_list_t             cpg_ctx_list;
 	/** actual number of items in CaRT contexts list */
-	int			cpg_ctx_num;
+	int                  cpg_ctx_num;
 	/** maximum number of contexts user wants to create */
-	uint32_t		cpg_ctx_max_num;
+	uint32_t             cpg_ctx_max_num;
 
 	/** free-list of indices */
-	bool			cpg_used_idx[CRT_SRV_CONTEXT_NUM];
+	bool                 cpg_used_idx[CRT_SRV_CONTEXT_NUM];
 
 	/** Hints to mercury/ofi for max expected/unexp sizes */
-	uint32_t		cpg_max_exp_size;
-	uint32_t		cpg_max_unexp_size;
+	uint32_t             cpg_max_exp_size;
+	uint32_t             cpg_max_unexp_size;
 
 	/** Number of remote tags */
-	uint32_t		cpg_num_remote_tags;
-	uint32_t		cpg_last_remote_tag;
+	uint32_t             cpg_num_remote_tags;
+	uint32_t             cpg_last_remote_tag;
 
 	/** Set of flags */
-	unsigned int		cpg_sep_mode		: 1,
-				cpg_primary		: 1,
-				cpg_contig_ports	: 1,
-				cpg_inited		: 1;
+	unsigned int    cpg_sep_mode : 1, cpg_primary : 1, cpg_contig_ports : 1, cpg_inited : 1;
 
 	/** Mutext to protect fields above */
-	pthread_mutex_t		cpg_mutex;
+	pthread_mutex_t cpg_mutex;
 };
 
 #define MAX_NUM_SECONDARY_PROVS 2
 
 /* CaRT global data */
 struct crt_gdata {
 	/** Providers iinitialized at crt_init() time */
-	int			cg_primary_prov;
-	int			cg_num_secondary_provs;
-	int			*cg_secondary_provs;
+	int                    cg_primary_prov;
+	int                    cg_num_secondary_provs;
+	int                   *cg_secondary_provs;
 
 	/** Provider specific data */
-	struct crt_prov_gdata	cg_prov_gdata_primary;
+	struct crt_prov_gdata  cg_prov_gdata_primary;
 
 	/** */
-	struct crt_prov_gdata	*cg_prov_gdata_secondary;
+	struct crt_prov_gdata *cg_prov_gdata_secondary;
 
 	/** global timeout value (second) for all RPCs */
-	uint32_t		cg_timeout;
+	uint32_t               cg_timeout;
 
 	/** global swim index for all servers */
-	int32_t			cg_swim_crt_idx;
+	int32_t                cg_swim_crt_idx;
 
 	/** credits limitation for #inflight RPCs per target EP CTX */
-	uint32_t		cg_credit_ep_ctx;
+	uint32_t               cg_credit_ep_ctx;
 
 	/** the global opcode map */
-	struct crt_opc_map	*cg_opc_map;
+	struct crt_opc_map    *cg_opc_map;
 	/** HG level global data */
-	struct crt_hg_gdata	*cg_hg;
+	struct crt_hg_gdata   *cg_hg;
 
-	struct crt_grp_gdata	*cg_grp;
+	struct crt_grp_gdata  *cg_grp;
 
 	/** refcount to protect crt_init/crt_finalize */
-	volatile unsigned int	cg_refcount;
+	volatile unsigned int  cg_refcount;
```

Comment:
(style)  Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt